processing/src/test/java/org/apache/druid/java/util/metrics/HttpPostEmitterMonitorTest.java

```diff
@@ -0,0 +1,129 @@
+
+    final String emitter_buffers_allocated_delta = "emitter/buffers/allocated/delta";
+    assertEquals(metricEvents.get(emitter_buffers_allocated_delta).get(0).getMetricEvent().getMetric(), emitter_buffers_allocated_delta);
+    assertEquals(metricEvents.get(emitter_buffers_allocated_delta).get(0).getMetricEvent().getValue(), 20);
+
+    final String emitter_batch_filling_max_time_ms = "emitter/batchFilling/maxTimeMs";
+    assertEquals(metricEvents.get(emitter_batch_filling_max_time_ms).get(0).getMetricEvent().getMetric(), emitter_batch_filling_max_time_ms);
+    assertEquals(metricEvents.get(emitter_batch_filling_max_time_ms).get(0).getMetricEvent().getValue(), 0);
+
+    final String emitter_buffers_dropped_delta = "emitter/buffers/dropped/delta";
+    assertEquals(metricEvents.get(emitter_buffers_dropped_delta).get(0).getMetricEvent().getMetric(), emitter_buffers_dropped_delta);
+    assertEquals(metricEvents.get(emitter_buffers_dropped_delta).get(0).getMetricEvent().getValue(), 10);
+
+    final String emitter_batch_filling_min_time_ms = "emitter/batchFilling/minTimeMs";
+    assertEquals(metricEvents.get(emitter_batch_filling_min_time_ms).get(0).getMetricEvent().getMetric(), emitter_batch_filling_min_time_ms);
+    assertEquals(metricEvents.get(emitter_batch_filling_min_time_ms).get(0).getMetricEvent().getValue(), 0);
+
+    final String emitter_events_emit_queue = "emitter/events/emitQueue";
+    assertEquals(metricEvents.get(emitter_events_emit_queue).get(0).getMetricEvent().getMetric(), emitter_events_emit_queue);
+    assertEquals(metricEvents.get(emitter_events_emit_queue).get(0).getMetricEvent().getValue(), 200l);
+
+    final String emitter_events_large_emit_queue = "emitter/events/large/emitQueue";
+    assertEquals(metricEvents.get(emitter_events_large_emit_queue).get(0).getMetricEvent().getMetric(), emitter_events_large_emit_queue);
+    assertEquals(metricEvents.get(emitter_events_large_emit_queue).get(0).getMetricEvent().getValue(), 75l);
+
+    final String emitter_buffers_reuse_queue = "emitter/buffers/reuseQueue";
+    assertEquals(metricEvents.get(emitter_buffers_reuse_queue).get(0).getMetricEvent().getMetric(), emitter_buffers_reuse_queue);
+    assertEquals(metricEvents.get(emitter_buffers_reuse_queue).get(0).getMetricEvent().getValue(), 15);
+
+    final String emitter_buffers_failed_delta = "emitter/buffers/failed/delta";
+    assertEquals(metricEvents.get(emitter_buffers_failed_delta).get(0).getMetricEvent().getMetric(), emitter_buffers_failed_delta);
+    assertEquals(metricEvents.get(emitter_buffers_failed_delta).get(0).getMetricEvent().getValue(), 5);
+
+    final String emitter_failed_sending_max_time_ms = "emitter/failedSending/maxTimeMs";
+    assertEquals(metricEvents.get(emitter_failed_sending_max_time_ms).get(0).getMetricEvent().getMetric(), emitter_failed_sending_max_time_ms);
+    assertEquals(metricEvents.get(emitter_failed_sending_max_time_ms).get(0).getMetricEvent().getValue(), 0);
+  }
+}
```

Comment:
Please add a newline at the end of the file.

Reply:
Done. Pushed in the next commit.